src/test/java/uk/gov/hmcts/divorce/systemupdate/schedule/SystemGenerateCurrentStateCountsReportTest.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.divorce.systemupdate.schedule;
+
+import com.google.common.collect.ImmutableList;
+import feign.FeignException;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.divorce.caseworker.service.notification.StateReportNotification;
+import uk.gov.hmcts.divorce.idam.IdamService;
+import uk.gov.hmcts.divorce.idam.User;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdConflictException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchCaseException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class SystemGenerateCurrentStateCountsReportTest {
+
+    @Mock
+    private CcdSearchService ccdSearchService;
+
+    @Mock
+    private IdamService idamService;
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Mock
+    private StateReportNotification reportNotificationService;
+
+    @InjectMocks
+    private SystemGenerateCurrentStateCountsReport reportTask;
+
+    private User user;
```

Comment:
Is it possible to mock this too?

Reply:
I donn't want to try returning a mock to a mock in a when